src/view/pages/Farmer/tabs/Info/index.tsx

```diff
@@ -0,0 +1,217 @@
+
+import createPage from '../../../../generators/Page/index';
+
+import styles from '../../styles';
+
+interface OwnStateType {
+  mode?: string;
+  selectedPaymentCycle: string;
+  selectedPaymentMethod: string;
+}
+
+interface OwnPropsType {
+  farmerFirstName: string;
+  farmerLastName: string;
+  farmerPhoneNumber: string;
+  farmerBusinessName: string;
+  farmerNotes: string;
+}
+
+interface DispatchPropsType {
+}
+
+interface StorePropsType {
+}
+
+type PropsType = OwnPropsType & DispatchPropsType & StorePropsType;
+
+/**
+ * Modes to determine how to render farmer information
+ */
+const modes = {
+  edit: 'edit',
+  view: 'view',
+};
+
+/**
+ * Container for application
+ */
+class FarmerInformation extends React.Component<PropsType, OwnStateType> {
+
+  constructor(props: OwnPropsType) {
+    super(props);
+    this.state = { 
+      mode: 'view',
+      selectedPaymentCycle: 'Weekly',
+      selectedPaymentMethod: 'Mobile',
+    };
+  }
+
+  private changeMode = (newMode: string) => {
+    this.setState(state => ({ mode: newMode }));
+  }
+
+  private updatePaymentCycle = (value: string) => {
+    this.setState(state => ({ selectedPaymentCycle: value }));
+  }
+  
+  private updatePaymentMethod = (value: string) => {
+    this.setState(state => ({ selectedPaymentMethod: value }));
+  }
+
+  /**
+   * Render method for Farmer Information
+   */
+  public render() {
+    if (this.state.mode === modes.view) {
+      return (
+        <Content padder>
+          <Grid>
+            <Row style={styles.farmerName}>
+              <H1>{this.props.farmerFirstName} {this.props.farmerLastName}</H1>
+            </Row>
+            <Row style={styles.infoLabel}>
+              <Label>Phone Number</Label>
+            </Row>
+            <Row style={styles.infoLabel}>
+              <H2>{this.props.farmerPhoneNumber}</H2>
+            </Row>
+            <Row style={styles.infoLabel}>
+              <Label>Business Name</Label>
+            </Row>
+            <Row style={styles.infoLabel}>
+              <H2>{this.props.farmerBusinessName}</H2>
+            </Row>
+            <Row style={styles.infoLabel}>
+              <Col>
+                <Label>Payment Cycle</Label>
+              </Col> 
+              <Col>
+                <Label>Payment Method</Label>
+              </Col>
+            </Row>
+            <Row style={styles.infoLabel}>
+              <Col>
+                <H2>{this.state.selectedPaymentCycle}</H2>
+              </Col> 
+              <Col>
+                <H2>{this.state.selectedPaymentMethod}</H2>
+              </Col>
+            </Row>
+            <Row style={styles.infoLabel}>
+              <Label>Notes</Label>
+            </Row>
+            <Row style={styles.infoLabel}>
+              <H2>{this.props.farmerNotes}</H2>
+            </Row>
+            <Row style={styles.editButton}>
+              <Col>
+                <Button block danger onPress={() => this.changeMode(modes.edit)}>
```

Comment:
This is a bad practice, and I don't believe that it actually passes our linter. Whereas in your stateless component before it's hard to optimize because it doesn't have an instance to store functions on, this is not so for a `ComponentClass`. `onPress` should receive a premade function, so it's not being recreated every single time we rerender. An example of this would be:

```
private changeToEditMode = () => this.changeMode('edit')
```

Note that with the strictness on `mode` that this is enforced by types, because if you put in `edit1` it would be a compilation error.